Improved a lot of things

Patitotective opened the pull request:
What the title says.
I should fix the conflicts...

Patitotective committed:
Improved a lot of things

Patitotective committed:
Fixed the installation instructions

Commit:
Delete config.nims

sharpcdf committed:
i just learned i hate PRs

Patitotective committed:
srcDir shouldn't exist when it's the root of the project

sharpcdf commented on README.md:
```diff
@@ -1,40 +1,61 @@
+## How it works?
+Jitter searches through GitHub for releases with `.tar.gz`, `.tgz` or `.zip` assets. Unlike Homebrew or similar package managers, jitter does not require a brewfile or nixfile in order to recognize the project.
 
+## Installing
```
good change, replacing the mug installer

sharpcdf commented on README.md:
```diff
@@ -1,40 +1,61 @@
+- `jtr update VSCodium/vscodium`: updates vscodium to the latest version.
+- `jtr update (this|jitter|jtr)`: updates jitter to the latest release.
+
+## About
```
remove :)

sharpcdf commented on README.md:
````diff
@@ -1,40 +1,61 @@
+nimble build
+```
+
+## Usage
````
you can just remove this, its the same info as ###examples

ghost commented:
put parenthesis at the end of all procs, with or without args for source code consistency

sharpcdf commented on jitter.nim:
```diff
@@ -1,237 +1,162 @@
+
+  success fmt"Successfully updated {pkg}"
+
+const parser = newParser:
```
add doc comments to the arg parser

sharpcdf commented on jitter/extract.nim:
```diff
@@ -0,0 +1,41 @@
+import std/[strformat, strutils, osproc, os]
```
add a var for `.jitter/nerve` in files that use it a lot such as this

sharpcdf commented on jitter/github.nim:
```diff
@@ -0,0 +1,126 @@
+  let data = content.parseJson()
+
+  if data.kind != JArray:
+    fatal fmt"Failed to fetch {pkg.gitFormat} releases"
```
change fetch to find

sharpcdf commented on jitter/github.nim:
```diff
@@ -0,0 +1,126 @@
+  try:
+    content = client.getContent(url)
+  except HttpRequestError:
+    fatal "Failed to fetch repositories"
```
fetch to find :)

sharpcdf commented on jitter/github.nim:
```diff
@@ -0,0 +1,126 @@
+    client.close()
+
+  let data = content.parseJson()
+  info &"Found {data[\"total_count\"]} repositories matching the query"
```
remove this, no need

sharpcdf commented on jitter/github.nim:
```diff
@@ -0,0 +1,126 @@
+  info &"Found {data[\"total_count\"]} repositories matching the query"
+
+  for repo in data["items"]:
+    list &"Github: {repo[\"full_name\"].getStr()}"
```
remove `list`, just add and return and then echo it from the caller location

sharpcdf commented on jitter/github.nim:
```diff
@@ -0,0 +1,126 @@
+    else:
+      fmt"https://api.github.com/repos/{pkg.owner}/{pkg.repo}/releases/tags/{pkg.tag}"
+
+  let client = newHttpClient(headers = newHttpHeaders([("accept", "application/vnd.github+json")]))
```
make this the global client and remove others

Patitotective replied:
That would create a client when `github.nim` is imported, that doesn't make any sense.

sharpcdf commented on jitter/github.nim:
```diff
@@ -0,0 +1,126 @@
+
+  extract(baseDir / "nerve" / downloadPath, pkg.dirFormat, make)
+
+proc ghDownload*(pkg: Package, make = true) =
```
what is this????

Patitotective replied:
It extracts the downloaded zip/tar to the package directory.

sharpcdf commented on jitter/github.nim:
```diff
@@ -0,0 +1,126 @@
+    list release["tag_name"].getStr()
+    result.add(release["tag_name"].getStr())
+
+proc searchGhRepo*(repo: string): seq[Package] = 
```
change `searchGhRepo` to just `ghSearch`

sharpcdf commented on jitter/github.nim:
```diff
@@ -0,0 +1,126 @@
+let baseDir = getHomeDir() & ".jitter/"
+let dDir = baseDir & "nerve/"
+
+proc listPkgGhReleases*(pkg: Package): seq[string] = 
```
change `listPkgGhReleases` to `ghListTags` or `ghListReleases`

sharpcdf reviewed (commented):
looks good, look at comments and make according changes :)

ghost commented:
@Patitotective made a review

ghost commented:
ill fix the conflicts afterwards

Patitotective committed:
- Created nerveDir and binDir in modules where they are needed
- Added comments for the arg parser
- Renamed some procs
- Removed about section in readme

Patitotective committed:
- Fixed some unupdated identifiers
- Added some check to see if version was provided

ghost commented:
some called procs still have no parenthesis in jitter.nim - not a big deal, i can change with the conflict fixes

sharpcdf committed:
made some fixes and changes

sharpcdf commented on jitter.nim:
```diff
@@ -1,237 +1,166 @@
+
+  let installedPkgs = getInstalledPkgs()
+
+  if not installedPkgs.anyIt(it.owner == pkg.owner and it.repo == pkg.repo):
```
fails to locate package

ghost commented:
i dont think you tested the changes after making them, i added the `config.nims` back, please make sure that the remove and other commands work first, i found search being inaccurate and remove not being able to find installed packages @Patitotective

sharpcdf commented on jitter/extract.nim:
```diff
@@ -0,0 +1,43 @@
+
+let baseDir = getHomeDir() & ".jitter/"
+let nerveDir = baseDir / "nerve"
+let binDir = baseDir / "bin"
```
add `bin/`, otherwise installs symlinks to parent directory with prefix `bin` (error)

sharpcdf replied:
might need to add same thing to `nerveDir`

Patitotective replied:
That cannot happen because the `/` procedure removes trailing slashes so it's essentially the same.

Patitotective replied:
That's not possible, the `/` proc removes trailing slashes.

Patitotective commented:
> i dont think you tested the changes after making them, i added the `config.nims` back, please make sure that the remove and other commands work first, i found search being inaccurate and remove not being able to find installed packages @Patitotective

Well, I really didn't change any of the logic of the original code, I just improved the code itself
But sure, I'll test them

Patitotective committed:
- Fixed empty names when searching
- Have to fix package format using __ because they are valid github identifiers

Patitotective committed:
__ fro ::

sharpcdf committed:
a little fix

sharpcdf committed:
bug fixes n' appimage support
